crates/anvil/src/eth/api.rs

```diff
@@ -2321,52 +2280,25 @@ impl EthApi {
                 block_env.clone(),
             );
 
-            // Exceptional case: init used too much gas, we need to increase the gas limit and try
-            // again
-            if let Err(BlockchainError::InvalidTransaction(InvalidTransactionError::GasTooHigh(
-                _,
-            ))) = ethres
-            {
-                // increase the lowest gas limit
-                lowest_gas_limit = mid_gas_limit;
-
-                // new midpoint
-                mid_gas_limit = (highest_gas_limit + lowest_gas_limit) / U256::from(2);
-                continue;
-            }
-
-            match ethres {
-                Ok((exit, _, _gas, _)) => match exit {
+            match ethres.try_into()? {
+                GasEstimationCallResult::Success(_) => {
                     // If the transaction succeeded, we can set a ceiling for the highest gas limit
                     // at the current midpoint, as spending any more gas would
                     // make no sense (as the TX would still succeed).
-                    return_ok!() => {
-                        highest_gas_limit = mid_gas_limit;
-                    }
-                    // If the transaction failed due to lack of gas, we can set a floor for the
-                    // lowest gas limit at the current midpoint, as spending any
-                    // less gas would make no sense (as the TX would still revert due to lack of
-                    // gas).
-                    InstructionResult::Revert |
-                    InstructionResult::OutOfGas |
-                    InstructionResult::OutOfFunds |
-                    // we're also checking for InvalidFEOpcode here because this can be used to trigger an error <https://github.com/foundry-rs/foundry/issues/6138> common usage in openzeppelin <https://github.com/OpenZeppelin/openzeppelin-contracts/blob/94697be8a3f0dfcd95dfb13ffbd39b5973f5c65d/contracts/metatx/ERC2771Forwarder.sol#L360-L367>
```

Comment:
can we keep this in the tryfrom impl

Reply:
With this PR overall logic changes and now any EVM error is ignored just as `InvalidFEOpcode`

Earlier we would throw a error on any revm error if it's not `Revert`, `OutOfGas` or `InvalidFEOpcode`, so the comment made sense, but I think now [this](https://github.com/foundry-rs/foundry/pull/7515/files#diff-a76a82dc700c0415fcd07ec10cde2ad56234e41af0ea3c07f8d1a8383b8b0913R2297-R2298) comment is enough to explain what's going on

Reply:
I see, then the comment is no longer useful, gotcha